### ISSUE TITLE
'show errors' is broken in 2018.2.36

### ISSUE THREAD

**Issue by pwillard2** [CoatiSoftware/SourcetrailBugTracker#577]
'show errors' does not work as expected any more.

For example, assuming I have an 'incomplete' file displayed in Code window, when I click on 'show errors' button (right of filename), it now shows some errors in Errors window (about 10 errors) AND the file displayed is changed to a different file (the file of 1st error displayed in Error window).
Very annoying...

In  previous Sourcetrail version, as I remember, the 1st click on 'show errors' would display all the errors in the Errors window, and a 2nd click would display the error of THAT source file only.
So NO CHANGE TO CURRENT FILE IN CODE WINDOW, and Errors window would display only the errors of that file only.

Basically 'show errors' should show errors of the current file (and not change the current file).

I am running 2018.2.36 - 64 bit on Centos.

**Comment by egraether** [comment]
With the `show errors` button we show all errors affecting the file. This means that in certain cases there are no errors in that file, only errors in files included by that file.

Therefore the `show errors` button answers this question: "Which errors do I need to fix to make this file complete?"

Which kind of behaviour do you suggest?

(Please take a look at the newest release again, there were some fixes related to errors)

**Comment by pwillard2** [comment]
I need to know why a specific file is not parsed completely (I see the file icon with a X on the left of the filename). At this point, I don't know how to get that info. I was expecting Show errors to show that information.
While browsing the code it's a MUST to be able to see all references, even if files that may not compile correctly.
First issue:  
- one file displayed in Code window, and has this Incomplete status. I click on 'show errors' and the error window does show that single error in that file (no fatal error). But it's not what I want. I want to fix the Incomplete status.
Second issue:
- this error window showing that single error, has both Fatals and Errors selected. (non-indexed not selected). Of course only 1 Error is seen. Now I de-select Fatals, and I see ALL Errors now. Weird. 
Third point:
I repeat this on a different file (also has Incomplete status). And I click on 'show errors'. And this time I see all errors for compiling that file, including 1 fatal error (in one include file, another include not found). So here no-error/no-fatal in the original file (AFAIU) but I see errors/fatals from included files. That's what I need I think for the 1st issue. 

(I am using version 2018.2.77 on Centos).

**Comment by egraether** [comment]
I think we have a misunderstanding of the definition of incomplete. A file is incomplete when:
* A: it has errors
* B: it was only indexed within translation units with fatal errors

Because of that, clicking `Show errors` might show you errors that are not within the same file. But if you fix those errors the file should be complete.

The second issue was a bug, fixed with 2018.2.83

**Comment by egraether** [comment]
New beta release is available and there where some related fixes. Can you please re-evaluate if this is working as expected for you?
[download beta](https://www.sourcetrail.com/downloads#beta)

**Comment by egraether** [comment]
No update in 3 months, closing
